File: client-scalaz/src/main/scala/org/openapitools/client/api/PlatformResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import PlatformResponse._

case class PlatformResponse (
  deviceId: Option[Long],
active: Option[Boolean],
name: Option[String],
display: Option[String],
minimum: Option[Long],
maximum: Option[Long],
downloadUrl: Option[String],
description: Option[String])

object PlatformResponse {
  import DateTimeCodecs._

  implicit val PlatformResponseCodecJson: CodecJson[PlatformResponse] = CodecJson.derive[PlatformResponse]
  implicit val PlatformResponseDecoder: EntityDecoder[PlatformResponse] = jsonOf[PlatformResponse]
  implicit val PlatformResponseEncoder: EntityEncoder[PlatformResponse] = jsonEncoderOf[PlatformResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/platform_response.clj
(ns sirqul-io-t-platform.specs.platform-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def platform-response-data
  {
   (ds/opt :deviceId) int?
   (ds/opt :active) boolean?
   (ds/opt :name) string?
   (ds/opt :display) string?
   (ds/opt :minimum) int?
   (ds/opt :maximum) int?
   (ds/opt :downloadUrl) string?
   (ds/opt :description) string?
   })

(def platform-response-spec
  (ds/spec
    {:name ::platform-response
     :spec platform-response-data}))
